apps/web/src/modules/dao/components/Feed/CastCard.tsx

```diff
@@ -0,0 +1,106 @@
+import { Box, Flex, Text } from '@zoralabs/zord'
+import axios from 'axios'
+import dayjs from 'dayjs'
+import relativeTime from 'dayjs/plugin/relativeTime'
+import React, { useMemo } from 'react'
+import useSWR from 'swr'
+
+import { CardSkeleton } from './CardSkeleton'
+import { cardWrapper } from './Feed.css'
+
+export const CastCard = ({
+  text,
+  fid,
+  timestamp,
+  hexHash,
+}: {
+  text: string
+  fid: number
+  timestamp: number
+  hexHash: string
+}) => {
+  const { data, error, isValidating } = useSWR(fid ? [fid] : undefined, () =>
+    axios
+      .get<{ displayName?: string; pfp?: string; fName?: string }>(
+        `/api/feed/userData?fid=${fid}`
+      )
+      .then((x) => x.data)
+  )
+
+  const time = useMemo(() => {
+    dayjs.extend(relativeTime)
+    const date = dayjs.unix(timestamp / 1000).fromNow()
+    return date
+  }, [timestamp])
+
+  if (error) {
+    // Design decision: how would we like to display a profile load fail?
+    return null
+  }
+  if (isValidating && !data) {
+    return <CardSkeleton />
+  }
+
+  return (
+    <Box
+      className={cardWrapper}
+      py={{ '@initial': 'x4', '@768': 'x6' }}
+      px={{ '@initial': 'x2', '@768': 'x6' }}
+      borderRadius={'phat'}
+      borderStyle={'solid'}
+      borderWidth={'normal'}
+      borderColor={'border'}
+      maxW={'x32'}
+      mb={'x6'}
+    >
+      <a
+        href={`https://warpcast.com/~/conversations/${hexHash}`}
+        target="_blank"
+        // className={cardLink}
+      >
+        <Flex align={'center'} mb={'x4'}>
+          <Box mr="x3" borderRadius="round">
+            <div
+              style={{
+                width: '32px',
+                height: '32px',
+                position: 'relative',
+                overflow: 'hidden',
+              }}
+            >
+              <img
```

Comment:
have we considered using next/image to resize to 32 x 32 here?

Reply:
Yeah, that was my first go-to. However, the next image API requires that I whitelist an origin. The challenge is that Farcaster has quite a few origins. 

I'm trying to track down their process, but it's hard to get a reply. So I used <img />  for now.

Reply:
could proxy the requests but we can discuss that in a followup

Reply:
ah yeah thanks for reminding me. makes sense. yeah we could def explore @neokry's suggestion as a separate feature